tmt/steps/execute/__init__.py

```diff
@@ -854,7 +786,20 @@ def extract_tmt_report_results(self, invocation: TestInvocation) -> list["tmt.Re
 
         collection.validate()
 
-        return self._process_results_reduce(invocation, collection.results)
+        # TODO: Can we use the Result.from_invocation(...) here? Should I somehow map the results
+        # to the test invocation here or not?
```

Comment:
Do we want to somehow map the subresults to the `TestInvocation`? I somehow feel there is still a space for the improvement in the code below. Any tips/recommendations?

Reply:
I *think* we should do that, but in a safe way - `from_test_invocation()` would change some things it shouldn't touch, like `name`, but also some very important things that need to be in order, like `serial-number`. Maybe we need is some kind of "sanitize" method that would do what was (is?) in `_process_results_partials()`:

```python
            # For the result representing the test itself, set the important
            # attributes to reflect the reality.
            if partial_result.name == test.name:
                partial_result.start_time = invocation.start_time
                partial_result.end_time = invocation.end_time
                partial_result.duration = invocation.real_duration
                partial_result.context = self.step.plan._fmf_context
```

I feel like I can't really focus at this moment, but this looks like an important step - we shouldn't probably mess with all names, but we should make sure timestamps and context and serial number are reported correctly, if nothing else. But then again, `from_test_invocation()` does exactly that, and if frameworks use it to construct their overall tests, we don't need to worry about it *here*.

Reply:
@seberm, @happz, I guess this one has been already covered?

Reply:
I would say yes, it's resolved (at least from my perspective). I don't think any further changes are needed and we can mark this as resolved.